File: Hash Table/0791.Custom Sort String/Solution.java
class Solution {
    public String customSortString(String order, String s) {
        int[] cnt = new int[26];
        for (char c : s.toCharArray()) {
            cnt[c - 'a']++;
        }
        StringBuilder ans = new StringBuilder();
        for (char c : order.toCharArray()) {
            while (cnt[c - 'a'] > 0) {
                cnt[c - 'a']--;
                ans.append(c);
            }
        }
        for (int i = 0; i < 26; i++) {
            while (cnt[i] > 0) {
                ans.append((char)('a' + i));
                cnt[i]--;
            }
        }
        return ans.toString();
    }
}

File: Hash Table/0791.Custom Sort String/Solution.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        cnt = [0] * 26
        for c in s:
            cnt[ord(c) - ord('a')] += 1
        n = len(s)
        ans = list()
        for c in order:
            while cnt[ord(c) - ord('a')] > 0:
                cnt[ord(c) - ord('a')] -= 1
                ans.append(c)
        for i in range(26):
            while cnt[i] > 0:
                ans.append(chr(ord('a') + i))
                cnt[i] -= 1
        return "".join(ans)
